public/app/features/dimensions/editors/NumberInput.tsx

```diff
@@ -45,8 +45,22 @@ export class NumberInput extends PureComponent<Props, State> {
   };
 
   onChange = (e: React.FocusEvent<HTMLInputElement>) => {
+    let newValue: string | undefined = undefined;
+    const min = this.props.min;
+    const max = this.props.max;
+    const currValue = e.currentTarget.valueAsNumber;
+
+    if (!Number.isNaN(currValue)) {
+      if (min && currValue < min) {
+        newValue = min.toString();
+      } else if (max && currValue > max) {
+        newValue = max.toString();
+      } else {
+        newValue = e.currentTarget.value;
+      }
+    }
     this.setState({
-      text: e.currentTarget.value,
+      text: newValue ? newValue : '',
```

Comment:
This approach definitely works in guarding user from entering any value less than or greater than max / min. It may be a little better UX though if we display a validation error message (yellow box form validation style) indicating to user that value entered was not in correct range. This would make it more explicit why the value is automatically updating. Thoughts?

Reply:
Sounds good - I ended up using our existing field validation format with an error message. It's not currently showing as yellow, but maybe down the line we can explore adding a warning style for inputs.